src/services/functionAppService.ts

```diff
@@ -90,27 +91,53 @@ export class FunctionAppService extends BaseService {
     Guard.null(functionApp);
 
     const getTokenUrl = `${this.baseUrl}${functionApp.id}/functions?api-version=2016-08-01`;
-    const response = await this.sendApiRequest("GET", getTokenUrl);
+    try {
+      const response = await Utils.runWithRetry(async () => {
+        const listFunctionsResponse = await this.sendApiRequest("GET", getTokenUrl);
 
-    if (response.status !== 200) {
+        if (listFunctionsResponse.status !== 200 || listFunctionsResponse.data.value.length === 0) {
+          this.log("-> Function App not ready. Retrying...");
+          throw new Error(listFunctionsResponse.data);
+        }
+
+        return listFunctionsResponse;
+      }, 10, 5000);
+
+      return response.data.value.map((functionConfig) => functionConfig.properties);
+    }
+    catch (e) {
+      this.log("-> Unable to retrieve function app list");
```

Comment:
we should exit with error if we cannot retrieve the list of functions

Reply:
Throughout the plugin so far, we've just logged the error and exited gracefully

Reply:
I agree with @mydiemho here.  If we are having a hard time getting the list of functions after deployment this can signal that there is some other slowdown which would likely cause next steps in the deployment process to fail.